app/Utils/Validation/HasValidator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Utils\Validation;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Validation\Rules\Unique;
+
+trait HasValidator
+{
+    abstract protected static function fieldValidator(string $field): string|array;
+
+    public static function getInsertValidator(array $fields): array
+    {
+        $ruleSet = [];
+        foreach ($fields as $field) {
+            $ruleSet[$field] = static::fieldValidator($field);
+        }
+        return $ruleSet;
+    }
+
+    public static function getPatchValidator(array $fields, Model|string $ignore): array
```

Comment:
w jakim przypadku chcemy tu stringa? `Model|string $ignore`

Reply:
Jak mam instancję Modelu żeby ją pominąć w Unique, to instancja Modelu

jak nie mamy instancji, tylko id w stringu, to można podać id w stringu

Reply:
ale zazwyczaj chcemy model